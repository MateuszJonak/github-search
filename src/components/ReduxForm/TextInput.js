/* @flow */

import React from 'react';
import TextField from '@material-ui/core/TextField';

type Props = {
  input: Object,
  meta: {
    touched: boolean,
    error: String,
  },
};

export default (props: Props) => {
  const {
    input,
    meta: { touched, error },
    ...inputProps
  } = props;

  const isError: boolean = touched && !!error;

  return (
    <TextField
      {...inputProps}
      onBlur={input.onBlur}
      onFocus={input.onFocus}
      onChange={input.onChange}
      value={input.value}
      error={isError}
      helperText={isError ? error : ''}
    />
  );
};
