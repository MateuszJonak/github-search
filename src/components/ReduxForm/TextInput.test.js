import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import TextInput from './TextInput';

describe('TextInput', () => {
  let props;
  let tree;
  const shallowEnzyme = properties => shallow(<TextInput {...properties} />);

  beforeEach(() => {
    props = {
      input: {
        onBlur: jest.fn(),
        onFocus: jest.fn(),
        onChange: jest.fn(),
        value: 'value',
      },
      meta: {
        touched: false,
        error: '',
      },
    };
    tree = shallowEnzyme(props);
  });

  it('should render component correctly with props', () => {
    expect(toJson(tree)).toMatchSnapshot();
  });

  it('should render component correctly with props error', () => {
    tree.setProps({
      meta: {
        touched: true,
        error: 'error',
      },
    });
    expect(toJson(tree)).toMatchSnapshot();
  });
});
