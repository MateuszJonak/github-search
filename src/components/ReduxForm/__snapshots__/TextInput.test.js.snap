// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInput should render component correctly with props 1`] = `
<TextField
  error={false}
  helperText=""
  onBlur={[Function]}
  onChange={[Function]}
  onFocus={[Function]}
  required={false}
  select={false}
  value="value"
/>
`;

exports[`TextInput should render component correctly with props error 1`] = `
<TextField
  error={true}
  helperText="error"
  onBlur={[Function]}
  onChange={[Function]}
  onFocus={[Function]}
  required={false}
  select={false}
  value="value"
/>
`;
